File: app/src/main/java/com/qhy040404/libraryonetap/utils/PermissionUtils.java
package com.qhy040404.libraryonetap.utils;

import android.app.Activity;
import android.content.pm.PackageManager;
import android.widget.Toast;

import androidx.core.app.ActivityCompat;

import com.qhy040404.libraryonetap.R;

public class PermissionUtils {
    private static void requestPermission(String[] s, Activity activity) {
        Toast.makeText(activity, R.string.promptPermission, Toast.LENGTH_SHORT).show();
        ActivityCompat.requestPermissions(activity, s, 100);
    }

    public boolean checkPermission(Activity activity, String[] permission) {
        boolean hasPermission = true;
        int j = 0;
        for (String s : permission) {
            if (activity.checkSelfPermission(s) == PackageManager.PERMISSION_DENIED) {
                if (s != null) {
                    hasPermission = false;
                    requestPermission(new String[]{s}, activity);
                }
            }
        }
        return hasPermission;
    }
}

File: app/src/main/java/com/qhy040404/libraryonetap/utils/PermissionUtils.kt
package com.qhy040404.libraryonetap.utils

import android.app.Activity
import android.content.pm.PackageManager
import androidx.annotation.StringRes
import androidx.fragment.app.FragmentManager
import com.qhy040404.libraryonetap.ui.fragment.fullscreen.FullScreenDialogFragment
import com.qhy040404.libraryonetap.utils.extensions.getString

object PermissionUtils {
  fun checkPermission(
    activity: Activity,
    permission: String,
    childFragmentMgr: FragmentManager,
    @StringRes strId: Int
  ): Boolean {
    return if (activity.checkSelfPermission(permission) == PackageManager.PERMISSION_DENIED) {
      FullScreenDialogFragment(strId.getString()).apply {
        show(childFragmentMgr, null)
      }
      activity.requestPermissions(arrayOf(permission), 100)
      false
    } else {
      true
    }
  }
}
